pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/v3/AdminApiTransactionTest.java

```diff
@@ -624,8 +624,9 @@ public void testUpdateTransactionCoordinatorNumber() throws Exception {
         }
 
         admin.transactions().scaleTransactionCoordinators(coordinatorSize * 2);
-        pulsarClient = PulsarClient.builder().serviceUrl(lookupUrl.toString()).enableTransaction(true).build();
+        replacePulsarClient(PulsarClient.builder().serviceUrl(lookupUrl.toString()).enableTransaction(true));
         pulsarClient.close();
```

Comment:
This kind of code read strange to me. What's the original purpose of such a code snippet? We obtain a new client but never use it, must we replace the global `pulsarClient` instead of init a local one?

Also, `replacePulsarClient` calls `shutdown` while we call `close` in those test code.

Reply:
@tisonkun I checked again and it's pretty clear what it does. If you remove the lines of code you will notice the impact. 

This is from the initTransaction method:
```java
        replacePulsarClient(PulsarClient.builder().serviceUrl(lookupUrl.toString()).enableTransaction(true));
        pulsarClient.close();
        pulsarClient = null;
        Awaitility.await().until(() ->
                pulsar.getTransactionMetadataStoreService().getStores().size() == coordinatorSize);
        replacePulsarClient(PulsarClient.builder().serviceUrl(lookupUrl.toString()).enableTransaction(true));
```
It will first close the existing pulsarClient instance and replace it with a pulsarClient instance that has transaction coordination client enabled.  The side-effect of the client connecting to the broker will initiate the required services in the broker. I don't know the exact reason to close the client and then create a new client, but that's how it is. It's not a part of this PR to start refactoring the test code beyond the purpose of fixing the Pulsar client leaks.

Reply:
I'm thinking of using a local PulsarClient instance to trigger the side-effect instead of replacing since the logical purpose has nothing to do wth "replace". But you're the author so you make the call of style :)